src/plugins/intel_cpu/src/shape_inference/static_dimension.cpp

```diff
@@ -94,17 +128,64 @@ bool StaticDimension::merge(StaticDimension& dst, const StaticDimension& d1, con
 }
 
 bool StaticDimension::broadcast_merge(StaticDimension& dst, const StaticDimension& d1, const StaticDimension& d2) {
-    if (d1 == 1) {
+    if (d1.get_length() == 1) {
         dst = d2;
         return true;
     }
-    if (d2 == 1) {
+    if (d2.get_length() == 1) {
         dst = d1;
         return true;
     }
     return merge(dst, d1, d2);
 }
 
+StaticDimension& StaticDimension::operator=(const ov::Dimension& dim) {
```

Comment:
What is happening there
Why so many new methods / operators are necessary to resolve this check?
And we still end up wit ha bunch of NOLINTs

Reply:
I guess one NOLINT with a reason will resolve the issue